### ISSUE TITLE
dynamic layouts

### ISSUE THREAD

**Issue by schochastics** [schochastics/graphlayouts#23]
[paper](https://kops.uni-konstanz.de/bitstream/handle/123456789/20924/Brandes_209246.pdf?sequence=2)

should be compliant with how `ggraph` and `gganimate` play together (see also #20)

**Comment by thomasp85** [comment]
FYI, I'm still not sure exactly how ggraph and gganimate should play together except for the most simple cases (edges coming and going) :-)

**Comment by schochastics** [comment]
I was thinking about that too and couldn't imagine how this should work. Speaking from a research perspective: Most dynamic networks will only fall into the simple case of edges coming and going over time for a fixed set of nodes. But I assume you want/need to be more general :). 

I guess some form of facet for dynamic networks would also be nice? Just to get the snapshots next to each other, like below done with `patchwork`.
![harry](https://user-images.githubusercontent.com/17147355/63251728-33061480-c266-11e9-979f-8322b9496f0b.png)

Anyway, the layout is there now and I am happy to change stuff (e.g. atm it requires a list of networks as input) to make an integration with ggraph/gganimate as smooth as possible.

**Comment by thomasp85** [comment]
Yeah, I'd prefer an interface to be as general as possible... I can imagine multiple scenarios that would need addressing:

- Fixed layout with edges coming and going (already works)
- Layout updating to changing topology (nodes moving around as edges come and go)
- Changing between different layout algorithms (mainly for illustrative purposes)

All of this also ties into better temporal network support in tidygraph

This is one of the things that won't make it into the next ggraph release, so there'll be some time to figure all this out
